--- ./startup/startup_stm32l433xx.s	2017-02-04 09:32:15.000000000 -0600
+++ ./Mobilinkd/TNC/startup_stm32l433xx.S	2017-03-10 09:20:38.137298430 -0600
@@ -96,6 +96,21 @@
 	adds	r2, r0, r1
 	cmp	r2, r3
 	bcc	CopyDataInit
+/* BEGIN BSS2 init code */
+    movs r1, #0
+    b  LoopCopyDataInit1
+ CopyDataInit1:
+     ldr  r3, =_sibss2
+     ldr  r3, [r3, r1]
+     str  r3, [r0, r1]
+     adds  r1, r1, #4
+ LoopCopyDataInit1:
+     ldr  r0, =_sbss2
+     ldr  r3, =_ebss2
+     adds  r2, r0, r1
+     cmp  r2, r3
+     bcc  CopyDataInit1
+/* END BSS2 init code */
 	ldr	r2, =_sbss
 	b	LoopFillZerobss
 /* Zero fill the bss segment. */
